### PR TITLE
Update docs for JS functions in web app

### LINKED ISSUE #243: Properly document javascript functions in web app code
We have a collection of several javascript functions, mainly located in `static/js/jwql.js`, that currently have minimal API documentation.  We should update these to have proper API documentation using `jsdocs` (http://usejsdoc.org/).

For example:

```javascript
/**
 * Determines what filetypes to use for the given thumbnail
 * @param {String} thumbnail_dir - The base location of the thumbnail
 * @param {Array} suffixes - An array of available suffixes for the observation of interest
 * @param {String} i - The index of the thumbnail
 * @param {String} file_root - The rootname of the observation of interest
 */
function determine_filetype_for_thumbnail(thumbnail_dir, suffixes, i, file_root) {

    // Update the thumbnail to show the most processed filetype
    var img = document.getElementById('thumbnail'+i);
    if (suffixes.indexOf("cal") >= 0) {
        var jpg_path = thumbnail_dir + file_root.slice(0,7) + '/' + file_root + '_cal_integ0.thumb';
        img.src = jpg_path;
    } else if (suffixes.indexOf("rate") >= 0) {
        var jpg_path = thumbnail_dir + file_root.slice(0,7) + '/' + file_root + '_rate_integ0.thumb';
        img.src = jpg_path;
    } else if (suffixes.indexOf("uncal") >= 0) {
        var jpg_path = thumbnail_dir + file_root.slice(0,7) + '/' + file_root + '_uncal_integ0.thumb';
        img.src = jpg_path;
    };
};
```

### PR DESCRIPTION
**Pull request by bourque**
This PR updates the API docs for the JavaScript functions within the web app code. 

Closes #243

### CODE REVIEW

File: jwql/website/apps/jwql/static/js/jwql.js [hunk 1]
```diff
@@ -250,9 +278,9 @@ function update_thumbnails_page(inst, proposal, base_url) {
         }});
 };
 
-/* JS function to construct the URL corresponding to a specific GitHub release
-* @param {String} version_string - The x.y.z version number
-*/
+/* Construct the URL corresponding to a specific GitHub release
+ * @param {String} version_string - The x.y.z version number
+ */
```
Comment (laurenmarietta) [inline jwql/website/apps/jwql/static/js/jwql.js]: Also looks like there are some extra spaces at the start of lines 282 and 283

File: jwql/website/apps/jwql/templates/view_image.html [hunk 2]
```diff
@@ -51,6 +55,11 @@
 
     	};
 
+
+        /**
+         * Change the integration number of the displayed image
+         * @param {String} direction - The direction to switch to, either "left" (descrease) or "right" (increase).
```
Comment (laurenmarietta) [inline jwql/website/apps/jwql/templates/view_image.html]: Another typo: "decrease"

File: jwql/website/apps/jwql/static/js/jwql.js [hunk 3]
```diff
@@ -82,6 +95,13 @@ function search() {
 };
 
 
+/**
+ * Limit the displayed thumbails based on filter criteria
```
Comment (laurenmarietta) [inline jwql/website/apps/jwql/static/js/jwql.js]: Typo: "thumbnails"

**Comment by laurenmarietta** [comment]
I just found a few typos (PyCharm is really good for that!), but besides those, this looks good.

Just out of curiosity: can these docstrings be included in our ReadTheDocs sometime?

File: jwql/website/apps/jwql/static/js/jwql.js (hunk 1 above)
Comment (bourque) [reply jwql/website/apps/jwql/static/js/jwql.js]: @laurenmarietta I think these are supposed to be there?

**Comment by bourque** [comment]
Thanks @laurenmarietta , another reason I should probably switch to PyCharm 😬 

Good question about ReadTheDocs.  I'm not sure if it is possible.  It would depend on if we could somehow make `sphinx` aware that these docstrings exist and tell it that they are some other format besides `numpydoc`.  I've never thought about this before!

File: jwql/website/apps/jwql/static/js/jwql.js (hunk 1 above)
Comment (bourque) [reply jwql/website/apps/jwql/static/js/jwql.js]: @laurenmarietta Actually this did make me realize there was a small discrepancy between this docstring and the others.

**Pull request merged**